cli/modules/modules.go

```diff
@@ -5,47 +5,109 @@ import (
+	"gopkg.in/yaml.v3"
 )
 
 const (
-	helpModuleName = "help"
+	manifestFileName = "manifest"
+	mainEntryPoint   = "main"
 )
 
-// ModuleInfo stores information about Tarantool CLI module.
-type ModuleInfo struct {
-	// Is this module internal (or external).
-	IsInternal bool
-	// Path to module (used only is module external).
-	ExternalPath string
+// Manifest stores information about Tarantool CLI module.
+type Manifest struct {
+	// Version of module.
+	Version string `yaml:"version"`
+	// Main is name of executable file.
+	Main string `yaml:"main"`
+	// Help is a short description of the module.
+	Help string `yaml:"help"`
+	// TtVersion is required a version of TT CLI (optional).
+	TtVersion string `yaml:"tt-version"`
+	// Description is a full description of the module (optional).
+	// It can be used in the future for the help command.
+	Description string `yaml:"description"`
+	// Homepage is a link to the module homepage (optional).
+	Homepage string `yaml:"homepage_url"`
 }
 
 // ModulesInfo stores information about all CLI modules.
-type ModulesInfo map[string]*ModuleInfo
+// Internal commands has the `nil` Manifest.
+type ModulesInfo map[string]*Manifest
+
+// modulesEntry keeps detected entry points while scan modules.
+type modulesEntry struct {
+	// Modules location path.
+	Directory string
+	// Path to manifest.yaml file.
+	Manifest string
+	// Path to `main` executable file.
+	Main string
+}
+
+// possibleModules map module name with found its entry points.
+type possibleModules map[string]modulesEntry
 
-// fillSubCommandsInfo collects information about subcommands.
-func fillSubCommandsInfo(cmd *cobra.Command, modulesInfo *ModulesInfo) {
+// fillInternalCommands collects information about subcommands.
+func fillInternalCommands(cmd *cobra.Command, modulesInfo *ModulesInfo) {
 	for _, subCmd := range cmd.Commands() {
 		commandPath := subCmd.CommandPath()
 		if _, found := (*modulesInfo)[commandPath]; !found {
-			(*modulesInfo)[commandPath] = &ModuleInfo{
-				IsInternal: true,
-			}
+			(*modulesInfo)[commandPath] = nil
 
 			if subCmd.HasSubCommands() {
-				fillSubCommandsInfo(subCmd, modulesInfo)
+				fillInternalCommands(subCmd, modulesInfo)
 			}
 		}
 	}
 }
 
+// readManifest parses the manifest file to module requirements.
+func readManifest(dir string, manifest string) (Manifest, error) {
+	mf := Manifest{}
+	data, err := os.ReadFile(manifest)
+	if err != nil {
+		return mf, fmt.Errorf("failed to read manifest: %s", err)
+	}
+	if err = yaml.Unmarshal(data, &mf); err != nil {
+		return mf, fmt.Errorf("failed to parse manifest: %s", err)
+	}
+	mf.Main, err = exec.LookPath(filepath.Join(dir, mf.Main))
+	if err != nil {
+		return mf, fmt.Errorf("failed to find module executable: %s", err)
+	}
+	if mf.Version == "" {
+		return mf, fmt.Errorf("version field is mandatory for module Manifest")
+	}
+	if mf.Help == "" {
+		return mf, fmt.Errorf("help field is mandatory for module Manifest")
+	}
+	return mf, nil
```

Comment:
We need to enable some linter...
```suggestion
func readManifest(dir string, manifest string) (Manifest, error) {
	mf := Manifest{}
	data, err := os.ReadFile(manifest)
	if err != nil {
		return mf, fmt.Errorf("failed to read manifest: %s", err)
	}

	if err := yaml.Unmarshal(data, &mf); err != nil {
		return mf, fmt.Errorf("failed to parse manifest: %s", err)
	}

	mf.Main, err = exec.LookPath(filepath.Join(dir, mf.Main))
	if err != nil {
		return mf, fmt.Errorf("failed to find module executable: %s", err)
	}

	if mf.Version == "" {
		return mf, fmt.Errorf("version field is mandatory for module Manifest")
	}
	if mf.Help == "" {
		return mf, fmt.Errorf("help field is mandatory for module Manifest")
	}

	return mf, nil
```